flake.nix

```diff
@@ -0,0 +1,57 @@
+{
+  description = "The Jakt Programming Language";
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-unstable;
+    flake-utils.url = github:numtide/flake-utils;
+    gitignore = {
+      url = github:hercules-ci/gitignore.nix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, gitignore }: let
+    forAllSystems = nixpkgs.lib.genAttrs flake-utils.lib.defaultSystems;
+    inherit (gitignore.lib) gitignoreSource;
+  in {
+    overlays.default = (final: prev:
+      let
+        buildInputs = with prev; [
+          clang
+          python3
+        ];
+        nativeBuildInputs = with prev; [
+          pkg-config
+          cmake
+          ninja
+       ];
+      in {
+        jakt = final.callPackage ./jakt.nix {
+          inherit gitignoreSource;
+          inherit buildInputs nativeBuildInputs;
+        };
+      }
+    );
+
+    packages = forAllSystems (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ self.overlays.default ];
+        };
+        jakt = pkgs.jakt;
+        default = jakt;
+      in rec {
+        inherit default jakt;
+      }
+    );
+    devShells = forAllSystems (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in {
+        default = pkgs.mkShell {
+          inherit (self.packages.${system}.default) buildInputs nativeBuildInputs;
+        };
```

Comment:
This is a lot of boilerplate. is it all necessary? Serenity's Toolchain/flake.nix is a lot simpler.